nomos-services/mempool/src/backend/mockpool/mod.rs

```diff
@@ -128,3 +198,37 @@ where
             .collect()
     }
 }
+
+impl<BlockId, Item, Key> RecoverableMempool for MockPool<BlockId, Item, Key>
+where
+    Key: Hash + Eq + Ord + Clone + Send,
+    Item: Clone + Send + 'static,
+    BlockId: Ord + Copy,
+{
+    type RecoveryState = Self;
+
+    fn recover(_settings: Self::Settings, state: Self::RecoveryState) -> Self {
+        state
+    }
+
+    fn save(&self) -> Self::RecoveryState {
+        self.clone()
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum Error {}
```

Comment:
Why and empty error? Maybe you want to use [`Infallible`](https://doc.rust-lang.org/std/convert/enum.Infallible.html) instead?